Assets/Scripts/BlackHoleConsumption.cs

```diff
@@ -3,20 +3,21 @@
 public class BlackHoleConsumption : MonoBehaviour
 {
 
+    public BlackHolePull blackHolePull;
+    
     private LevelManager levelManager;
-    private BlackHolePull blackHolePull;
 
     private void Start()
     {
         levelManager = FindObjectOfType<LevelManager>();
-        blackHolePull = GetComponentInParent<BlackHolePull>();
     }
 
     private void OnTriggerEnter(Collider other)
     {
         // TODO: only consume when the colliding object is FULLY within the collider
```

Comment:
Instead of "fully in the collider", you should just start a "kill timer" when they enter the consumption collider. After the timer finishes, the body is actually consumed.

Reply:
Good idea. It only really matters for destroying non-player objects like asteroids, so I just called Destroy() with a 1 second delay